frontends/infinite-corridor/src/pages/DemoPage.tsx

```diff
@@ -0,0 +1,188 @@
+import React, { useState, useCallback } from "react"
+import Container from "@mui/material/Container"
+
+import {
+  useCourseSearch,
+  buildSearchQuery,
+  SearchQueryParams,
+  Aggregation
+} from "@mitodl/course-search-utils"
+import { LearningResourceSearchResult, SearchInput } from "ol-search-ui"
+
+import axios from "../libs/axios"
+import { useHistory } from "react-router"
+
+const pageSize = 5
+
+interface Result {
+  _source: LearningResourceSearchResult
+}
+
+const SEARCH_API_URL = "search/"
+
+const search = async (params: SearchQueryParams) => {
+  const body = buildSearchQuery(params)
+  try {
+    const { data } = await axios.post(SEARCH_API_URL, body)
+    return data
+  } catch (err) {
+    return null
+  }
+}
+
+type ResultsListProps = {
+  results: Result[]
+}
+
+const ResultsList: React.FC<ResultsListProps> = ({ results }) => {
+  return (
+    <div>
+      {results.map(hit => (
+        <div
+          className="resource-result"
+          key={hit._source.object_type.concat(hit._source.id.toString())}
+        >
+          {hit._source.title}{" "}
+          {hit._source.offered_by && hit._source.offered_by.length > 0 ?
+            `(${hit._source.offered_by.join(", ")})` :
+            null}
+        </div>
+      ))}
+    </div>
+  )
+}
+
+const DemoPage: React.FC = () => {
+  const [courseResults, setCourseResults] = useState<Result[]>([])
+  const [videoResults, setVideoResults] = useState<Result[]>([])
+  const [podcastResults, setPodcastResults] = useState<Result[]>([])
+
+  const [completedInitialLoad, setCompletedInitialLoad] = useState(false)
+  const [requestInFlight, setRequestInFlight] = useState(false)
+  const [searchApiFailed, setSearchApiFailed] = useState(false)
+
+  const clearSearch = useCallback(() => {
+    setCourseResults([])
+    setVideoResults([])
+    setPodcastResults([])
+    setCompletedInitialLoad(false)
+  }, [])
+
+  const runSearch = useCallback(async (text: string) => {
+    if (text) {
+      setRequestInFlight(true)
+      const from = 0
+      const courseFilter = { type: ["course"] }
+      const videoFilter = { type: ["video"] }
+      const podcastFilter = { type: ["podcast"] }
+      const newCourseResults = await search({
+        text,
+        from,
+        activeFacets: courseFilter,
+        size:         pageSize
+      })
+
+      const newVideoResults = await search({
+        text,
+        from,
+        activeFacets: videoFilter,
+        size:         pageSize
+      })
+
+      const newPodcastResults = await search({
+        text,
+        from,
+        activeFacets: podcastFilter,
+        size:         pageSize
+      })
+
+      setRequestInFlight(false)
+
+      if (
+        !newCourseResults ||
+        newCourseResults["apiFailed"] ||
```

Comment:
Can the results include "apiFailed" ?

Reply:
I'm not sure where this is coming from. We check for it in ocw-hugo-themes as well https://github.com/mitodl/ocw-hugo-themes/pull/662/files#diff-b681b498395009bcd9a1b7c4c214db07e6997fd551aec850c8a026447a7e70f2R66